pkg/apiserver/resthandler.go

```diff
@@ -29,26 +30,20 @@ import (
 	"github.com/emicklei/go-restful"
 )
 
-// ResourceNameFunc returns a name (and optional namespace) given a request - if no name is present
-// an error must be returned.
-type ResourceNameFunc func(req *restful.Request) (namespace, name string, err error)
-
-// ObjectNameFunc  returns the name (and optional namespace) of an object
-type ObjectNameFunc func(obj runtime.Object) (namespace, name string, err error)
-
-// ResourceNamespaceFunc returns the namespace associated with the given request - if no namespace
-// is present an error must be returned.
-type ResourceNamespaceFunc func(req *restful.Request) (namespace string, err error)
-
-// LinkResourceFunc updates the provided object with a SelfLink that is appropriate for the current
-// request.
-type LinkResourceFunc func(req *restful.Request, obj runtime.Object) error
+// ScopeNamer handles accessing names from requests and objects
+type ScopeNamer interface {
```

Comment:
Exported types need documentation. Each of the exported methods in the interface do as well.